content/design/vendoring.md

```diff
@@ -53,3 +53,224 @@ already written):
 You now have a `vendor` directory that contains the sources for `goagen`, compile them and use the
 tool as described above. Don't forget to recompile the tool after each `glide update` that updates
 `goa`.
+
+# Code-based generation and vendoring
+
+> For the impatients, here is a concise version of what we'll see here:
+>
+> ```
+> $ mkdir -p $GOPATH/src/github.com/your/pkg/design
+> $ cd !$
+> $ # Create your design in the `design` package
+> $ # Paste de content of `gen/main.go` below to `...github.com/your/pkg/gen/main.go`
+> $ go run gen/main.go      # generates all the things
+> $ govendor init           # creates the vendor/ dir
+> $ govendor add +e         # pull needed packages in
+> ```
+>
+> And you're set. Everyone in your team will be able to deterministically
+> regenerate your code if they tweak the design, without version conflicts.
+
+Instead of using `goagen` to get started, you can write code that'll
+do the same, but as it will mark import paths in code, your prefered
+vendoring tool will be able to pick up any dependencies by following
+the dependency graph of the code:
+
+```
+cd $GOPATH/src/github.com/your/pkg
+mkdir gen
+```
+
+Then put this content into `gen/main.go`:
+
+```
+package main
+
+import (
+	_ "github.com/your/pkg/design"
+
+	"github.com/goadesign/goa/design"
+	"github.com/goadesign/goa/goagen/codegen"
+	genapp "github.com/goadesign/goa/goagen/gen_app"
+	genclient "github.com/goadesign/goa/goagen/gen_client"
+	genjs "github.com/goadesign/goa/goagen/gen_js"
+	genmain "github.com/goadesign/goa/goagen/gen_main"
+	genschema "github.com/goadesign/goa/goagen/gen_schema"
+	genswagger "github.com/goadesign/goa/goagen/gen_swagger"
+)
+
+func main() {
+	codegen.ParseDSL()
+	codegen.Run(
+		&genmain.Generator{
+			API:    design.Design,
+			Target: "app",
+		},
+		&genswagger.Generator{
+			API: design.Design,
+		},
+		&genapp.Generator{
+			API:    design.Design,
+			OutDir: "app",
+			Target: "app",
+			NoTest: true,
+		},
+		&genclient.Generator{
+			API: design.Design,
+		},
+		&genschema.Generator{
+			API: design.Design,
+		},
+		&genjs.Generator{
+			API: design.Design,
+		},
+	)
+}
+```
+
+Remove the generators you don't need from the `Run()` call.
+
+Merely by having that little code in place, all the `goa` code
+generation code will be vendored in.
```

Comment:
Maybe instead of the above:

The code above will cause all the `goa` code generation packages to be vendored appropriately.